src/redux/investment/actionCreator.js

```diff
@@ -0,0 +1,89 @@
+import { SERVER_URL, API_URL, API_KEY } from 'react-native-dotenv';
+import {
+  FETCH_USER_INVESTMENT_SHARES_START,
+  FETCH_USER_INVESTMENT_SHARES_FAIL,
+  FETCH_USER_INVESTMENT_SHARES_SUCCESS,
+  FETCH_STOCK_PRICE_START,
+  FETCH_STOCK_PRICE_FAIL,
+  FETCH_STOCK_PRICE_SUCCESS,
+  FETCH_STOCK_PRICE,
+} from './actionType';
+import { round, addComma } from '../../common/numbers';
+
+export const fetchUserInvestmentSharesStart = () => ({
+  type: FETCH_USER_INVESTMENT_SHARES_START,
+});
+
+export const fetchUserInvestmentSharesFail = (payload) => ({
+  type: FETCH_USER_INVESTMENT_SHARES_FAIL,
+  payload,
+});
+
+export const fetchUserInvestmentSharesSuccess = (payload) => ({
+  type: FETCH_USER_INVESTMENT_SHARES_SUCCESS,
+  payload,
+});
+
+export const fetchStockPriceStart = () => ({
+  type: FETCH_STOCK_PRICE_START,
+});
+
+export const fetchStockPriceFail = (payload) => ({
+  type: FETCH_STOCK_PRICE_FAIL,
+  payload,
+});
+
+export const fetchStockPriceSuccess = (payload) => ({
+  type: FETCH_STOCK_PRICE_SUCCESS,
+  payload,
+});
+
+export const fetchStockPrice = (stocks, symbols) => (dispatch) => {
+  const apiUrl = new URL(`${API_URL}/stock/market/batch`);
+  apiUrl.searchParams.append('symbols', symbols);
+  apiUrl.searchParams.append('types', 'price');
+  apiUrl.searchParams.append('token', API_KEY);
+  dispatch(fetchStockPriceStart());
+  fetch(apiUrl)
+    .then((response) => {
+      if (!response.ok) {
+        throw new Error('Oops, there\'s something wrong with our app.');
+      }
+      return response.json();
+    })
+    .then((response) => {
+      dispatch(fetchStockPriceSuccess(response));
+      const calculatedValue = stocks
+        .map(({ shares, symbol }) => shares * response[symbol].price)
+        .reduce((a, b) => a + b);
+      dispatch({
+        type: FETCH_STOCK_PRICE,
+        payload: addComma(round(calculatedValue)),
+      });
+    })
+    .catch((err) => dispatch(fetchStockPriceFail(err)));
+};
+
+export const displayInvestmentsValue = (uid) => (dispatch) => {
```

Comment:
- fetchInvestmentsValue
- also, it is better to hardcode the user id here, since when we will use the access token, we will read it from the store here, not in the component